File: GCD/GCD/ViewController.m
//
//  ViewController.m
//  GCD
//
//  Created by shishir roy on 28/10/17.
//  Copyright © 2017 shishir roy. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()
@property (weak, nonatomic) IBOutlet UIImageView *imageView;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

-(void) viewDidAppear:(BOOL)animated{
    dispatch_queue_t queVar = dispatch_queue_create("my first queue", NULL);
    
    //dispatch_async(<#dispatch_queue_t  _Nonnull queue#>, <#^(void)block#>)
    
    dispatch_async(queVar, ^{
        NSLog(@"Hi How are u....??");
        NSString *url = @"http://microblogging.wingnity.com/JSONParsingTutorial/will.jpg";
        NSData *data = [NSData dataWithContentsOfURL:[NSURL URLWithString:url]];
        
        dispatch_async(dispatch_get_main_queue(), ^{
                _imageView.image = [UIImage imageWithData:data];
        });
    });
}

@end

File: iOS/JSONParser/JSONParser/ViewController.swift
//
//  ViewController.swift
//  JSONParser
//
//  Created by shishir roy on 30/10/17.
//  Copyright © 2017 shishir roy. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        // optional chaning
        
        //let path = Bundle.main.path(forResource: "usersAPI", ofType: "txt")
        //print(path ?? "Not a real path")
        
        guard let path = Bundle.main.path(forResource: "usersAPI", ofType: "txt") else {return}
        let url = URL(fileURLWithPath: path)
        //let data = try! Data(contentsOf: url)
        
        do {
            let data = try Data(contentsOf: url)
            print(data)
            //JSON Serialization of data
            let json = try JSONSerialization.jsonObject(with: data, options: .mutableContainers)
            print(json)
            
        } catch  {
            print(error)
        }
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

